godot-macros/src/class/derive_godot_class.rs

```diff
@@ -253,6 +253,73 @@ fn make_godot_init_impl(class_name: &Ident, fields: &Fields) -> TokenStream {
     }
 }
 
+fn make_onready_init(all_fields: &[Field]) -> TokenStream {
+    let mut onready_fields = all_fields
+        .iter()
+        .filter(|&field| field.is_onready)
+        .map(|field| {
+            let field = &field.name;
+            quote! {
+                ::godot::private::auto_init(&mut self.#field, &base);
+            }
+        });
+
+    if let Some(first) = onready_fields.next() {
+        quote! {
+            {
+                let base = <Self as godot::obj::WithBaseField>::to_gd(self).upcast();
+                #first
+                #( #onready_fields )*
+            }
+        }
+    } else {
+        TokenStream::new()
+    }
```

Comment:
Why is only the first field read here?

Or what's the point of mapping/filtering the whole slice and then discard all but one element?